### ISSUE TITLE
Not able to 

### ISSUE THREAD

**Issue by vinaymcp** [christocracy/cordova-plugin-background-fetch#29]
Hello,

I installed the plugin in my IOS project, everything got installed properly, but now the app is not calling up the services every 15 mins in background mode.  So what i did was in the debug mode, i tried to use 'simulate background fetch', but then i started to get a error as 

# "Warning: Application delegate received call to -application:performFetchWithCompletionHandler: but the completion handler was never called"

I have enabled background service in the app settings.

By any chance do i need a apple account to make this work  sorry for dumb questions? 

Any clarifications will help!

Following is my code 

CGRect screenBounds = [[UIScreen mainScreen] bounds];
    [[UIApplication sharedApplication]
     setMinimumBackgroundFetchInterval:
     UIApplicationBackgroundFetchIntervalMinimum];  //  [[UIApplication sharedApplication] setMinimumBackgroundFetchInterval:1.0];

-(void) application:(UIApplication *)application performFetchWithCompletionHandler:(void (^)(UIBackgroundFetchResult))completionHandler
{
   
    NSLog(@"################back calld###################");
      completionHandler(UIBackgroundFetchResultNoData);
    
   
}

----------------------------------------------------------------------
**Controller : **
----------------------------------------------------------------------
ionic.Platform.ready(function(){
    // will execute when device is ready, or immediately if the device is already ready.
    var Fetcher = window.backgroundFetch;

        alert(5);
        var startTime="17:03:46";
        var enddate="17:03:46";
        var resultedtime="";
        var resultedtotime="";
        alert(6);
        // Your background-fetch handler.
        var fetchCallback = function() {
             alert(1);
            console.log('BackgroundFetch initiated');
          
           // Fetcher.finish(); 
            // perform your ajax request to server here
           $.get({
            url: '/heartbeat.json',
            
            callback: function(response) {
                // process your response and whatnot.
    
                Fetcher.finish();   // <-- N.B. You MUST called #finish so that native-side can signal completion of the background-thread to the os.
            }
            });
        }
          Fetcher.configure(fetchCallback);
          
          });

Kindly help, struggling from last 1 week to get this working.

Thanks in advance!!!

**Comment by christocracy** [comment]
Never use JavaScript alert method while debugging a background process.  It
is a *blocking* method.  That is, your code halts after:

  alert(1):

In the callback.  The subsequent lines of code won't be executed
(Fetcher.finish() in particular) until you press the [ok] button on the
alert dialog.

On Monday, March 14, 2016, vinaymcp <notifications@github.com
<javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:

> Hello,
>
> I installed the plugin in my IOS project, everything for installed
> properly, but now the app is not calling up the services every 15 mins in
> background mode. So in debug mode, i tried to use simulate background
> fetch, but then i started to get a error as
> "Warning: Application delegate received call to
> -application:performFetchWithCompletionHandler: but the completion handler
> was never called"
>
> I have enabled background service in the settings.
>
> Following is my code
>
> CGRect screenBounds = [[UIScreen mainScreen] bounds];
> [[UIApplication sharedApplication]
> setMinimumBackgroundFetchInterval:
> UIApplicationBackgroundFetchIntervalMinimum]; // [[UIApplication
> sharedApplication] setMinimumBackgroundFetchInterval:1.0];
>
> -(void) application:(UIApplication *)application
> performFetchWithCompletionHandler:(void
> (^)(UIBackgroundFetchResult))completionHandler
> {
>
> NSLog(@"################back calld###################");
>   completionHandler(UIBackgroundFetchResultNoData);
>
> }
> ------------------------------
> **Controller : **
>
> ionic.Platform.ready(function(){
> // will execute when device is ready, or immediately if the device is
> already ready.
> var Fetcher = window.backgroundFetch;
>
>     alert(5);
>     var startTime="17:03:46";
>     var enddate="17:03:46";
>     var resultedtime="";
>     var resultedtotime="";
>     alert(6);
>     // Your background-fetch handler.
>     var fetchCallback = function() {
>          alert(1);
>         console.log('BackgroundFetch initiated');
>
>        // Fetcher.finish();
>         // perform your ajax request to server here
>        $.get({
>         url: '/heartbeat.json',
>
>         callback: function(response) {
>             // process your response and whatnot.
>
>             Fetcher.finish();   // <-- N.B. You MUST called #finish so that native-side can signal completion of the background-thread to the os.
>         }
>         });
>     }
>       Fetcher.configure(fetchCallback);
>
>       });
>
> Kindly help, struggling from last 1 week to get this working.
>
> Thanks in advance!!!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/christocracy/cordova-plugin-background-fetch/issues/29>
> .
>


-- 
Snet form Gmail Mobile

**Comment by vinaymcp** [comment]
I am getting the same issue even after removing the alert

CDVBackgroundFetch onFetch
2016-03-14 15:48:52.637 sample[1463:390663] Warning: Application delegate received call to -application:performFetchWithCompletionHandler: but the completion handler was never called.

**Comment by vinaymcp** [comment]
the above issue is solved....

But the problem still remains that the code executes only once , but it does not get executed for second time.

I am having this code in the controller

ionic.Platform.ready(function(){
// will execute when device is ready, or immediately if the device is already ready.
var Fetcher = window.backgroundFetch;

    alert(5);
    var startTime="17:03:46";
    var enddate="17:03:46";
    var resultedtime="";
    var resultedtotime="";
    alert(6);
    // Your background-fetch handler.
    var fetchCallback = function() {
         alert(1);
        console.log('BackgroundFetch initiated');

       // Fetcher.finish(); 
        // perform your ajax request to server here
       $.get({
        url: '/heartbeat.json',

        callback: function(response) {
            // process your response and whatnot.

            Fetcher.finish();   // <-- N.B. You MUST called #finish so that native-side can signal completion of the background-thread to the os.
        }
        });
    }
      Fetcher.configure(fetchCallback);

      });

Thanks

**Comment by christocracy** [comment]
If "above issue is solved", then this issue issue is closed.  If you have further problems, open a new issue.
